From e404d4726c62c837801887632d2ba13dbd77ff1e Mon Sep 17 00:00:00 2001
From: Alex Kiernan <alex.kiernan@hivehome.com>
Date: Sun, 22 Oct 2017 20:36:36 +0000
Subject: [PATCH] post-update-check: Merge changes from refkit-core

post-update checks fail in upstream, so cherry pick changes from Intel
---
 Makefile.am                               | 9 +++++++--
 hooks/run-hooks                           | 9 +++++++++
 services/refkit-update-post-check.service | 2 +-
 3 files changed, 17 insertions(+), 3 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 9ae7000..6f69eab 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -38,9 +38,14 @@ install-data-local:
 	    fi; \
 	done
 
-	install -m 0755 -d $(DESTDIR)$(HOOK_DIR)/post-update.d
+	install -m 0755 -d $(DESTDIR)$(HOOK_DIR)/post-update-check.d
 	install -m 0755 -T $(top_srcdir)/hooks/run-hooks \
-	    $(DESTDIR)$(HOOK_DIR)/post-update
+	    $(DESTDIR)$(HOOK_DIR)/post-update-check
+	for h in $(top_srcdir)/hooks/post-update-check.d/[0-9]*; do \
+	    if [ -x $$h ]; then \
+	        install -m 0755 $$h $(DESTDIR)$(HOOK_DIR)/post-update-check.d; \
+	    fi; \
+	done
 
 	install -m 0755 -d $(DESTDIR)$(HOOK_DIR)/rollback.d
 	install -m 0755 -T $(top_srcdir)/hooks/run-hooks \
diff --git a/hooks/run-hooks b/hooks/run-hooks
index 039b09c..c298684 100644
--- a/hooks/run-hooks
+++ b/hooks/run-hooks
@@ -1,12 +1,21 @@
 #!/bin/sh
 
+if [ $# = 0 -a "${0##*/}" = "post-update-check" ]; then
+    read prev curr < /var/.ostree.updated
+    set $prev $curr
+fi
+
 HOOK_DIR=$0.d
+HOOK_TYPE=${0##*/}
 
 for h in $HOOK_DIR/[0-9]*-*; do
     if [ -e $h -a -x $h ]; then
+        hook=${h##*/}
+        echo "Executing $HOOK_TYPE hook $hook..."
         $h $*
         status=$?
         if [ $status != 0 ]; then
+            echo "hook $hook failed."
             exit $?
         fi
     fi
diff --git a/services/refkit-update-post-check.service b/services/refkit-update-post-check.service
index b80fa29..5f12835 100644
--- a/services/refkit-update-post-check.service
+++ b/services/refkit-update-post-check.service
@@ -5,7 +5,7 @@ ConditionPathExists=/var/.ostree.updated
 
 [Service]
 Type=oneshot
-ExecStart=/usr/bin/refkit-ostree-update --post-update-check
+ExecStart=/usr/share/refkit-ostree/hooks/post-update-check
 # OnFailure=ostree-post-update-failure.target
 RemainAfterExit=Yes
 TimeoutSec=0
-- 
2.7.4

